pkg/controllers/clusterresource/keymanagementprovider_controller.go

```diff
@@ -37,33 +44,94 @@ type KeyManagementProviderReconciler struct {
 	Scheme *runtime.Scheme
 }
 
-func (r *KeyManagementProviderReconciler) ReconcileWithConfig(ctx context.Context, config map[string]interface{}) (ctrl.Result, error) {
+func (r *KeyManagementProviderReconciler) ReconcileWithType(ctx context.Context, req ctrl.Request, refresherType string) (ctrl.Result, error) {
+	logger := logrus.WithContext(ctx)
+
+	var resource = req.Name
```

Comment:
QQ: Shall we use `resource := req.Name` here

Reply:
Excellent catch, you have a keen eye @junczhu! Yes `:=` should be used here.